src/SLCore/Analysis/SLCoreAnalyzer.cs

```diff
@@ -122,15 +122,15 @@ [new FileUri(path)],
         }
     }
 
-    private Dictionary<string, string> GetExtraProperties(IEnumerable<AnalysisLanguage> detectedLanguages)
+    private Dictionary<string, string> GetExtraProperties(string path, IEnumerable<AnalysisLanguage> detectedLanguages)
     {
         Dictionary<string, string> extraProperties = [];
         if (!IsCFamily(detectedLanguages))
         {
             return extraProperties;
         }
 
-        var compilationDatabasePath = compilationDatabaseLocator.Locate();
+        var compilationDatabasePath = compilationDatabaseLocator.GetOrNull(path);
         if (compilationDatabasePath != null)
```

Comment:
Currently, it seems that if a C++ file without a project is opened, we try to trigger the analysis without setting the `cfamily.compile-commands` property. In the previous implementation, analysis is skipped IIUC:

https://github.com/SonarSource/sonarlint-visualstudio/blob/6f4ca55dcd8ccdbff04985dd0b9668700f83c90d/src/CFamily/Analysis/CLangAnalyzer.cs#L127-L130

This currently results in the following confusing log message after this change:

```
[ThreadId 41] [DEBUG] [SLCORE] Start analysis
[ThreadId 38] [DEBUG] [SLCORE] Index files
[ThreadId 38] [DEBUG] [SLCORE] Language of file "file:///C:/Users/user/source/repos/Project3/Project3/Folder/Test.cpp" is detected to be "CPP"
[ThreadId 38] [DEBUG] [SLCORE] 1 file indexed
[ThreadId 71] [DEBUG] [SLCORE] Execute Sensor: CFamily
[ThreadId 41] [DEBUG] [SLCORE] CFamily plugin version: 6.60.0.76379 (5be0f75cd1a7285a862a55f79e785691e99cc32b)
[ThreadId 88] [SLCORE] 

The only way to get an accurate analysis of C/C++/Objective-C files in Manual Configuration mode is to provide a compilation database through the property "sonar.cfamily.compile-commands"; The option was not specified.

You can generate a compilation database by wrapping your clean build with SonarSource build-wrapper, or by using a third party tool.
For more information consult the documentation at https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/languages/c-family/analysis-modes/

If you don't want to analyze C/C++/Objective-C files, then prevent them from being analyzed by setting the following properties:

    sonar.c.file.suffixes=-
    sonar.cpp.file.suffixes=-
    sonar.objc.file.suffixes=-


[ThreadId 34] [SLCORE] Error executing sensor: 'CFamily'
```

Whereas previously, it was more clear:

```
[ThreadId 14] [DEBUG] [CompilationDatabaseLocator] No CMakeSettings file was found under C:\Users\user\source\repos\Project2, returning default location: C:\Users\user\source\repos\Project2\out\build\No Configurations\compile_commands.json
[ThreadId 14] [CMake] Could not locate compilation database at 'C:\Users\user\source\repos\Project2\out\build\No Configurations\compile_commands.json'. Make sure that your project is configured correctly. 
    See https://docs.sonarsource.com/sonarlint/visual-studio/getting-started/requirements/#language-specific-requirements for more information..
[ThreadId 14] Unable to retrieve the configuration for file 'C:\Users\user\source\repos\Project2\Project2\Folder\Test.cpp'.
    Check the file is part of a supported project type in the current solution.
```

Do you think it is possible to pass the equivalent of the scanner properties mentioned above when we don't expect the CFamily sensor to process the file?

Reply:
Thinking about it, I think it might be better to just build a single-file compilation database using the configuration of any file in the project.

This would be consistent with the experience when using a CMake compilation database, where we have such fallback logic (see [here](https://github.com/SonarSource/sonar-cpp/blob/4e48dbba0b6489010f5bf1d3ab5e5bdf7d1a53f7/sonar-cfamily-plugin/src/main/java/com/sonar/cpp/plugin/SonarLintSensor.java#L181-L203)) and by extension, the experience on VSCode. It also seems to be consistent with how CLion tries to guess configurations for files outside the project.

Let me know what you think...

Reply:
This is an interesting point. In this initial implementation, we decided to not reject the analysis of such files because of the secrets analysis. Rationale being that even if CFamily analysis doesn't work for the file, we would still like to show secrets in it. Do you think that is a bad idea?

Reply:
I will create a ticket for this and tackle it in a separate PR. If you have any comments, add them here: https://sonarsource.atlassian.net/browse/SLVS-1669

Reply:
> Rationale being that even if CFamily analysis doesn't work for the file, we would still like to show secrets in it. Do you think that is a bad idea?

I think the new implementation's approach makes sense. As I mentioned in my previous comment, if we can still create a compdb with the configuration for any file in the same project, it would make the user experience closer to that of CMake projects and other IDEs.

Thanks for creating the ticket, I will add my proposal there.

Reply:
It seems that we also hit this path in CMake mode when we fail to locate the generated compilation database file (thanks to @mostafa-mohammed-sonarsource for discovering it).

This can happen, for example, when the `set(EXPORT_COMPILE_COMMANDS ON)` is missing (which we instruct users to do in [our docs](https://docs.sonarsource.com/sonarqube-for-ide/visual-studio/getting-started/requirements/#getting-started-vs2019-and-later)) or when unsupported CMakePresets is used (#2612).

I would suggest opening a different ticket for this, where we have a similar set of alternative options to evaluate (except that we can't generate a temporary compilation database on our side for this case). WDYT? @georgii-borovinskikh-sonarsource